client/src/components/EditReviewForm.jsx

```diff
@@ -0,0 +1,115 @@
+import { useEffect, useState } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { fetchParks, getUserProfileInfo, newReview } from '../utils/utils';
+import '../styles/EditReviewForm.css';
+import { useNavigate, Link } from 'react-router';
+import Rating from '@mui/material/Rating';
+
+const EditReviewForm = () => {
+    const [parkData, setParkData] = useState([]);
+    const [selectedPark, setSelectedPark] = useState(null);
+    const [rating, setRating] = useState(1);
+    const [review, setReview] = useState('');
+    const { user } = useAuth();
+    const [userReviews, setUserReviews] = useState([]);
+    const nav = useNavigate();
+
+    const handleParkChange = (e) => {
+        const id = e.target.value;
+        setSelectedPark(id);
+    };
+
+    const handleFormSubmit = async (e) => {
+        e.preventDefault();
+        const reviewData = {
+            locationId: selectedPark,
+            rating,
+            review,
+        };
+        const response = await newReview(reviewData);
+        nav('/profile');
+    };
+
+    useEffect(() => {
+        const getUserRole = async () => {
+            if (user) {
+                const userData = await getUserProfileInfo(user.id);
+                setUserReviews(userData.reviews);
+            }
+        };
+        getUserRole();
+    }, [user]);
+
+    useEffect(() => {
+        fetchParks((fetchedParks) => {
+            setParkData(fetchedParks);
+            if (fetchedParks.length > 0) {
+                const parkId = fetchedParks[0].id;
+                setSelectedPark(parkId.toString());
+            }
+        });
+    }, []);
+
+    useEffect(() => {
+        if (userReviews.length > 0 && selectedPark) {
+            const selectedReview = userReviews.find(
+                (review) => review.locationId.toString() === selectedPark
+            );
+            if (selectedReview) {
+                setRating(parseInt(selectedReview.rating));
+                setReview(selectedReview.review);
+            } else {
+                setRating(1);
+                setReview('');
+            }
+        }
```

Comment:
i think you can move this logic out of `useEffect`, rating and review doesn't need to be states.

        if (userReviews.length > 0 && selectedPark) {
            const selectedReview = userReviews.find(
                (review) => review.locationId.toString() === selectedPark
            );
            if (selectedReview) {
                rating = parseInt(selectedReview.rating);
                review = selectedReview.review;
            } else {
                rating = 1;
                review = '';
            }
        }